radio/src/gui/colorlcd/module_setup.cpp

```diff
@@ -288,6 +288,34 @@ void ModuleWindow::updateModule()
     }
   }
 
+#if defined(INTERNAL_MODULE_PXX1) && defined(EXTERNAL_ANTENNA)
+  if (moduleIdx == INTERNAL_MODULE && isModuleXJT(moduleIdx) &&
+      g_eeGeneral.antennaMode == ANTENNA_MODE_PER_MODEL) {
+    auto line = newLine(&grid);
+    new StaticText(line, rect_t{}, STR_ANTENNA, 0, COLOR_THEME_PRIMARY1);
+    auto antennaChoice = new Choice(
+        line, rect_t{}, STR_ANTENNA_MODES, ANTENNA_MODE_INTERNAL,
+        ANTENNA_MODE_EXTERNAL,
+        GET_DEFAULT(g_model.moduleData[INTERNAL_MODULE].pxx.antennaMode),
+        [=](int32_t antenna) -> void {
```

Comment:
For code readability, you probably want to have this `Choice` be a specific component which has this code as a method of its own class (inherited from `Choice`).